pkg/daemon/ovs_linux.go

```diff
@@ -339,36 +339,41 @@ func configureContainerNic(nicName, ifName string, ipAddr, gateway string, isDef
 		}
 
 		if gwCheckMode != gatewayModeDisabled {
-			underlayGateway := gwCheckMode == gatewayCheckModeArping
+			var (
+				underlayGateway = gwCheckMode == gatewayCheckModeArping || gwCheckMode == gatewayCheckModeArpingNotConcerned
+				interfaceName   = nicName
+			)
 
-			if u2oInterconnectionIP != "" {
-				if nicType != util.InternalType {
-					if err := waitNetworkReady(ifName, ipAddr, u2oInterconnectionIP, false, true); err != nil {
-						return err
-					}
-				} else {
-					if err := waitNetworkReady(nicName, ipAddr, u2oInterconnectionIP, false, true); err != nil {
-						return err
-					}
-				}
+			if nicType != util.InternalType {
+				interfaceName = ifName
 			}
 
-			if nicType != util.InternalType {
-				return waitNetworkReady(ifName, ipAddr, gateway, underlayGateway, true)
+			if u2oInterconnectionIP != "" {
+				if err := checkGatewayReady(gwCheckMode, interfaceName, ipAddr, u2oInterconnectionIP, false, true); err != nil {
+					return err
+				}
 			}
-			return waitNetworkReady(nicName, ipAddr, gateway, underlayGateway, true)
+			return checkGatewayReady(gwCheckMode, interfaceName, ipAddr, gateway, underlayGateway, true)
 		}
 
 		return nil
 	})
 }
 
-func waitNetworkReady(nic, ipAddr, gateway string, underlayGateway, verbose bool) error {
+func checkGatewayReady(gwCheckMode int, intr, ipAddr, gateway string, underlayGateway, verbose bool) error {
+	if gwCheckMode == gatewayCheckModeArpingNotConcerned || gwCheckMode == gatewayCheckModePingNotConcerned {
+		go waitNetworkReady(intr, ipAddr, gateway, underlayGateway, verbose, 1)
+		return nil
```

Comment:
The waitNetworkReady with only one probe should return fast enough. No need to run in a goroutine and it will lead the lint error